src/common/database.c

```diff
@@ -2099,22 +2100,48 @@ static void _sanitize_db(dt_database_t *db)
-                                                db->error_dbfilename, db->error_message ? db->error_message : "");
-
-    dt_gui_show_standalone_yes_no_dialog(_("darktable - error locking database"), label_text, _("close darktable"),
+  { 
+    char lck_pathname[1024];
+    snprintf(lck_pathname, sizeof(lck_pathname), "%s.lock", db->error_dbfilename);
+    char *lck_dirname = g_strdup(lck_pathname);
+    char *lck_filename = g_strrstr(lck_dirname, "/") + 1 ;
+    *g_strrstr(lck_dirname, "/") = '\0';
+
+    char *label_text = g_markup_printf_escaped(
+        _("\n"
+          " At startup, the database failed to open because at least one of the two files in the database is locked. \n"
+          "\n"
+          " The persistence of the lock is mainly caused by one of the two following causes: \n"
+          " - Another occurrence of darktable has already opened this database file and locked it for its benefit. \n"
+          " - A previous occurrence of darktable ended abnormally and therefore \n"
+          "   could not close one or both files in the database properly. \n"
+          "\n"
+          " How to solve this problem? \n"
+            " 1 - Search in your environment if another darktable occurrence is active. If so, use it or close it. \n"
+            "     The lock indicates that the process number of this occurrence is : <i><b>%d</b></i> \n"
+            " 2 - If you can't find this other occurrence, try closing your session and reopening it or shutting down your computer. \n"
+            "     This will delete all running programs and thus close the database correctly. \n"
+            " 3 - If these two actions are not enough, it is because at least one of the two files that materialize the locks remains \n"
+            "     and that these are no longer attached to any occurrence of darktable. It is then necessary to delete it (or them). \n"
+            "     The two files are named <i>data.db.lock</i> and <i>library.db.lock</i> respectively. The opening mechanism signals \n"
+            "     the presence of the <i><b>%s</b></i> file in the <i><b>%s</b></i> folder \n"
+            "     (full pathname: <i><b>%s</b></i>). \n"
+            "     <u>Caution!</u> Do not delete these files without first checking that there are no more occurrences of darktable, \n"
+            "     otherwise you risk generating serious inconsistencies in your database. \n"
+          "\n"
+          " As soon as you have identified and removed the cause of the lock, darktable will start without any problem. \n"),
+      db->error_other_pid, lck_filename, lck_dirname, lck_pathname);
+
+    dt_gui_show_standalone_yes_no_dialog(_("darktable cannot be started because the database is locked."), label_text, _("close darktable"),
                                          /*_("try again")*/NULL);
 
+    g_free(lck_dirname);
     g_free(label_text);
   }
 
   g_free(db->error_message);
   g_free(db->error_dbfilename);
+  //db->error_other_pid = 0;
```

Comment:
Please remove commented out code. Or comment if really needed.

Reply:
Sorry, I commented on it because it didn't compile. To come back to it later and then I forgot.